### ISSUE TITLE
aws_eip public_dns resource is incorrect for China elastic IP's

### ISSUE THREAD

**Issue by heffergm** [hashicorp/terraform-provider-aws#9609]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
terraform -v
Terraform v0.11.14
+ provider.aws v2.22.0
+ provider.powerdns v0.1.0
```

### Affected Resource(s)

* aws_eip.*.public_dns

### Behavior

When using the `aws_eip.*.public_dns` attribute, the computed public_dns names for China resources are incorrect, in that they lack the `.cn` suffix.

As an example, below what you see is the computed resource record change when terraform applying without the fix we've put in place to add the `.cn` suffix to the resources. Note that the correct resource record is being removed and replaced with one that isn't valid for China (again, missing the `.cn` suffix).

```
      records.2466334667: "" => "ec2-xx-xx-xxx-xxx.cn-north-1.compute.amazonaws.com." (forces new resource)
      records.3351082987: "ec2-xx-xx-xxx-xxx.cn-north-1.compute.amazonaws.com.cn." => "" (forces new resource)
```

Since we don't see this same issue using the `aws_instance.*.public_dns` attribute, it would appear to be a bug specific to `aws_eip`.

**Comment by ewbankkit** [comment]
We should now be able to use the partition's `DNSSuffix` attribute as https://github.com/aws/aws-sdk-go/pull/2711 has been released in [AWS SDK v1.21.9](https://github.com/aws/aws-sdk-go/releases/tag/v1.21.9).

Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/9555

**Comment by ewbankkit** [comment]
@heffergm Thanks for raising this issue.
It should have been resolved via https://github.com/hashicorp/terraform-provider-aws/pull/12189.
